plugins/net_plugin/net_plugin.cpp

```diff
@@ -2801,20 +2800,20 @@ namespace eosio {
          return;
       }
 
-      auto ptrx = std::make_shared<transaction_metadata>( trx, my_impl->chain_plug->chain().configured_subjective_signature_length_limit() );
```

Comment:
Before the `check_variable_sig_size` check was done here and it would throw an exception out of this function. Now that check is delayed until key recovery and I'm not sure how the exception will be handled.

It isn't clear whether this is actually a problem. It depends on what the point of that validation was and when it is expected to be done.

Reply:
The check is now in `producer_plugin::on_incoming_transaction_async` where the transaction_metadata is created via a future. The `future.get()` here https://github.com/EOSIO/eos/pull/7721/files#diff-a2a21b62383e9cc5c5349676661db760R422 will pass the exception on to `next()` which will log it. Before the exception would have been handled in `net_plugin::process_next_message` which would have `close()` the connection which was probably not what was intended.